dali/dalistop/dalistop.cpp

```diff
@@ -45,26 +49,45 @@ int main(int argc, char* argv[])
             CMessageBuffer mb;
             int fn=-1;
             mb.append(fn);
-            if (comm->verifyConnection(0,2000)) {
+            if (comm->verifyConnection(0,2000))
+            {
                 comm->send(mb,0,MPTAG_DALI_COVEN_REQUEST,MP_ASYNC_SEND);
                 if (nowait)
+                {
                     Sleep(1000);
+                    exitCode = 0;
+                }
                 else
-                    while (comm->verifyConnection(0,1000)) {
-                        PROGLOG("Waiting for Dali Server to stop....");
-                        Sleep(5000);
+                {
+                    // verifyConnection() has a min conn timeout of 10s
+                    // use recv() instead to check for socket closed ...
+                    try
+                    {
+                        while (comm->recv(mb,0,MPTAG_DALI_COVEN_REQUEST,nullptr,1000))
```

Comment:
I misread, this should say:
```
    while (!comm->recv(mb,0,MPTAG_DALI_COVEN_REQUEST,nullptr,1000))
```

i.e. keep trying to receive until get something (the cancel notification) or an exception e.g. mp link closed.